api/proto/certs/certs.proto

```diff
@@ -0,0 +1,57 @@
+// Copyright(c) 2017-2018 Zededa, Inc.
+// All rights reserved.
+
+syntax = "proto3";
+
+option go_package  = "github.com/lf-edge/eve/api/go/certs";
+option java_package = "com.zededa.cloud.uservice.proto";
+
+// This is the request payload for POST /api/v1/edgeDevice/EdgeNodeCerts
+// or /api/v2/edgeDevice/EdgeNodeCerts
+// ZEdgeNodeCertMsg carries a set of X.509 certificate and their properties
+// from EVE to Controller.
+// The message is assumed to be protected by a TLS session bound to the
+// device certificate for v1.
+// The message is assumed to be protected by signing envelope for v2.
+message ZEdgeNodeCertMsg {
+  string deviceUuid = 1;
+  ZCertMsgTypes ztype = 2;
+  repeated ZCert certs = 3;
+}
+
+// This is the response payload for GET /api/v1/edgeDevice/ControllerCerts
+// or /api/v2/edgeDevice/ControllerCerts
+// ZEdgeNodeCertMsg carries a set of X.509 certificate and their properties
+// from Controller to EVE.
+message ZControllerCertMsg {
+  ZCertMsgTypes ztype = 1;
+  repeated ZCert certs = 2;
+}
+
+message ZCert {
+   string certId = 1;   //Identifier for the cert, usually a short hash of the cert.
+   ZCertType type = 2;  //what kind of certificate(to identify the target use case)
+   string cert = 3;  //X509 cert in .PEM format
+   ZCertProperties properties = 4;
+}
+
+enum ZCertType {
+   CERT_TYPE_DEVICE_ONBOARDING = 0;
+   CERT_TYPE_DEVICE_RESTRICTED_SIGNING = 1;
+   CERT_TYPE_DEVICE_ENDORSEMENT_RSA = 2;
+   CERT_TYPE_DEVICE_ECDH_EXCHANGE = 3;
+   CERT_TYPE_CONTROLLER_SIGNING = 4;
+}
+
+message ZCertProperties {
+   bool isMutable = 1; //if this is set, this certificate can't be overwritten.
+}
+
+//
+// The request type: tells what to do with the certs being sent.
+//
+enum ZCertMsgTypes {
+   CERT_MSG_TYPE_DEVICE_CERT = 0;
+   CERT_MSG_TYPE_ECO = 1;
```

Comment:
Which ECO? Underspecified.

Reply:
removed ECO. We can add new types when we have a case.